docs: add explanation for providedIn any (#35283)
angular/angular: aio/content/guide/providers.md

Though you can provide services by lazy loading modules, not all services can be lazy loaded. For instance, some modules only work in the root module, such as the Router. The Router works with the global location object in the browser.

> With Angular 9, it is possible to provide a new instance of service with each lazy loaded module. You can change the UserService code like:

Reviewer: ```suggestion
With Angular version 9, you can provide a new instance of service with each lazy loaded module. The change to `UserService` would be as follows:
```

Reviewer 2: Done.

> With Angular version 9, you can provide a new instance of service with each lazy loaded module. The change to `UserService` would be as follows:

Reviewer: Oops! I must have missed this one:
```suggestion
With Angular version 9, you can provide a new instance of a service with each lazy loaded module. The change to `UserService` would be as follows:
```

> With Angular version 9, you can provide a new instance of a service with each lazy loaded module. The change to `UserService` would be as follows:

Reviewer 3: ```suggestion
With Angular version 9, you can provide a new instance of a service with each lazy loaded module. The following code adds this functionality to `UserService`.
```

<code-example path="providers/src/app/user.service.2.ts"  header="src/app/user.service.ts"></code-example>

> Remember this is available for User defined services, and all eager loaded module will share an singleton insance, as shown in image below.

Reviewer: ```suggestion
Using `providedIn: 'any'` is available for user defined services, or services.... . 
With this configuration, all eagerly loaded modules share a singleton instance, however lazy loaded modules each get their own unique instance, as in the following diagram.
```
Could you define user defined services?

Reviewer 2: Done and removed this text to avoid confusion.

> With `providedIn: 'any'`, all eagerly loaded modules share a singleton instance, however lazy loaded modules each get their own unique instance, as in the following diagram.

Reviewer 3: ```suggestion
With `providedIn: 'any'`, all eagerly loaded modules share a singleton instance; however, lazy loaded modules each get their own unique instance, as shown in the following diagram.
```

> Using `providedIn: 'any'` is available for provider scope, With this configuration, all eagerly loaded modules share a singleton instance, however lazy loaded modules each get their own unique instance, as in the following diagram.

Reviewer: How about:

> With `providedIn: 'any'`, all eagerly loaded modules share a singleton instance, however lazy loaded modules each get their own unique instance, as in the following diagram.

Reviewer 2: Made some changes to image and changes the text.
